Automattic/jetpack
# CRM Automations UI: Add Edit Modal

Fixes https://github.com/Automattic/zero-bs-crm/issues/3213
Fixes https://github.com/Automattic/zero-bs-crm/issues/3212
Fixes https://github.com/Automattic/zero-bs-crm/issues/3204
Fixes https://github.com/Automattic/zero-bs-crm/issues/3215

## Proposed changes:


This PR adds an edit modal for the Automations UI.

<img width="598" alt="image" src="https://github.com/Automattic/jetpack/assets/42627630/23486008-af46-4aed-aa8f-92ec0b868e70">

### Other information:


## Jetpack product discussion


NtYgueI8oAgiWGHTbHpnjh-fi-1_29

## Does this pull request change what data or activity we track or use?



## Testing instructions:






1. Review the code.
2. Put add_filter( 'jetpack_crm_feature_flag_automations', '__return_true' ); in an mu-plugin.
3. Visit /wp-admin/admin.php?page=jpcrm-automations#/automations/test-edit-modal. Click the button which opens the modal. Note that saving is not implemented, loading is not truly implemented. The existing loading code is just a placeholder. Compare visuals to what is found in the Figma. (Note that this control isn't in the Figma so I made it up based on existing styles in the Figma. Feel free to make any style suggestions.)
